### PR TITLE
modal component modifications

### PR DESCRIPTION
**Pull request by lfergon**

### CODE REVIEW

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 1]
```diff
@@ -20,38 +20,56 @@ var NewAppModalComponent = React.createClass({
...
       onDestroy: $.noop
     };
   },
-
+ 
   getInitialState: function () {
     return {
       model: new App(),
+      parametersCount: 0,
+      volumesCount: 0,
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Nice to set default state here!

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /** @jsx React.DOM */
-
+
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: What's this space?

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 3]
```diff
@@ -158,21 +181,26 @@ var NewAppModalComponent = React.createClass({
...
+ 
     var generalErrors = errors.filter(function (e) {
-        return (e.attribute === "general");
-      });
-
+      return (e.attribute === "general");
+    });
+ 
+    var inputs = [];
+    for (var i=0;i<this.state.parametersCount; i++) {
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Please with spaces.
```
for (var i = 0; i < this.state.parametersCount; i++) {
```

Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Why do you fill an array with numbers?

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 4]
```diff
@@ -158,21 +181,26 @@ var NewAppModalComponent = React.createClass({
...
+      inputs.push(i);
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Why do you fill an array with numbers?

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 5]
```diff
@@ -259,9 +287,105 @@ var NewAppModalComponent = React.createClass({
                 errors={errors}>
               <input />
             </FormGroupComponent>
+            <hr />
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Don't use a hr-element for styling and separation. You could do separation by wrapping an element around. Line drawing could be done by CSS borders.

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 6]
```diff
@@ -259,9 +287,105 @@ var NewAppModalComponent = React.createClass({
...
+            {inputs.map(function (result) {
+              return <FormGroupComponent
+                attribute="key"
+                label="Key"
+                model={model}
+                errors={errors}>
+                <input />
+              </FormGroupComponent>;
+            })}
+            <button className="btn btn-default" type="button" onClick={this.addParameters}>
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Over 80 chars in width, please break down.
```
<button className="btn btn-default"
    type="button"
    onClick={this.addParameters}>
```

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 7]
```diff
@@ -259,9 +287,105 @@ var NewAppModalComponent = React.createClass({
...
+              +
+            </button>
+            <button className="btn btn-default" type="button" onClick={this.removeParameters}>
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Breaking 80 char rule.

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 8]
```diff
@@ -259,9 +287,105 @@ var NewAppModalComponent = React.createClass({
...
+                label="Mode"
+                model={model}
+                errors={errors}>
+              <select>
+                <option value="read">Read only</option>
+                <option value="write">Write - Read</option>
+                <option value="admin">Admin</option>
+              </select>
+            </FormGroupComponent>
+            <button className="btn btn-default" type="button" onClick={this.addVolumes}>
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Breaking 80 char rule.

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 9]
```diff
@@ -259,9 +287,105 @@ var NewAppModalComponent = React.createClass({
...
+              +
+            </button>
+            <button className="btn btn-default" type="button" onClick={this.removeVolumes}>
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Breaking 80 char rule.

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 10]
```diff
@@ -271,5 +395,5 @@ var NewAppModalComponent = React.createClass({
     );
   }
 });
-
-module.exports = NewAppModalComponent;
+ 
+module.exports = NewAppModalComponent;
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: No required newline at the end.

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 11]
```diff
@@ -132,9 +150,14 @@ var NewAppModalComponent = React.createClass({
     if ("instances" in modelAttrs) {
       modelAttrs.instances = parseInt(modelAttrs.instances, 10);
     }
-
+    if ("privileges" in modelAttrs) {
+      if (modelAttrs.privileges === "true") {
+        modelAttrs.instances = true;
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Shouldn't be modelAttrs.instances here?

You could do this without if statement:
```
modelAttrs.privileges = modelAttrs.privileges === "true"
```

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 12]
```diff
@@ -259,9 +287,105 @@ var NewAppModalComponent = React.createClass({
...
+              <input />
+            </FormGroupComponent>
+            <FormGroupComponent
+                attribute="value"
+                label="Value"
+                model={model}
+                errors={errors}>
+              <input />
+            </FormGroupComponent>
+            {inputs.map(function (result) {
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: What is this doing? No value field there..

And you should create a function from it to create the list directly from state.parameterCount.

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 13]
```diff
@@ -158,21 +181,26 @@ var NewAppModalComponent = React.createClass({
...
   render: function () {
     var model = this.state.model;
     var errors = this.state.errors;
-
+ 
     var generalErrors = errors.filter(function (e) {
-        return (e.attribute === "general");
-      });
-
+      return (e.attribute === "general");
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: No need for parentheses here.

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 14]
```diff
@@ -259,9 +287,105 @@ var NewAppModalComponent = React.createClass({
...
+              +
+            </button>
+            <button className="btn btn-default" type="button" onClick={this.removeVolumes}>
+              -
+            </button>
             <div>
               {errorBlock}
-              <input type="submit" className="btn btn-success" value="+ Create" /> <button className="btn btn-default" type="button" onClick={this.destroy}>
+              <input type="submit" className="btn btn-success" value="+ Create" /> 
+              <button className="btn btn-default" type="button" onClick={this.destroy}>
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Breaking 80 char rule.

File: src/main/resources/assets/js/components/NewAppModalComponent.jsx [hunk 15]
```diff
@@ -20,38 +20,56 @@ var NewAppModalComponent = React.createClass({
...
+ 
   destroy: function () {
     // This will also call `this.props.onDestroy` since it is passed as the
     // callback for the modal's `onDestroy` prop.
     this.refs.modalComponent.destroy();
   },
-
+ 
+  addParameters: function () {
+    this.setState({parametersCount: this.state.parametersCount + 1});
```
Comment (aldipower) [inline src/main/resources/assets/js/components/NewAppModalComponent.jsx]: Nice setting state here!

**Comment by aldipower** [comment]
Please describe what this pull request should do and tries to implement and reference to the according issue.

**Comment by aldipower** [comment]
Not functioning.
Missing styling at all.

**Comment by lfergon** [comment]
Buttons for + and - are totally a mess. I need to align them with the input fields. I can not find icons/style for round buttons on main.less or in bootstrap.css 
I removed the code related to add new row to the table, was a bad decision, now I have a better understanding of how could work. I need to improve A LOT my knowledge of React

**Comment by lfergon** [comment]
Show/hide settings and show/hide docker options have not the correct style.

**Comment by drexin** [comment]
@aldipower What's the status of this? Should we close it?

**Comment by kolloch** [comment]
@aldipower we close this now if you do not object.

**Pull request closed without merging**